addons/project/controllers/portal.py

```diff
@@ -24,6 +25,42 @@ def _prepare_portal_layout_values(self):
         values['task_count'] = Task.search_count([('project_id', 'in', projects.ids)])
         return values
 
+    # ------------------------------------------------------------
+    # My Project
+    # ------------------------------------------------------------
+    def _project_check_access(self, project_id, access_token=None):
```

Comment:
@tde-banana-odoo Should we generate this method on the portal controller class ? We have `_invoice_check_access` , and `_order_check_access` , now we are adding `_project_check_access` and `_purchase_check_access`. The implentation is always the same. Passing the model name as a parameter can reduce code size.
```
    def _check_document_access(self, model_name, document_id, access_token=None):
        document = request.env[model_name].browse(document_id)
        document_sudo = document.sudo()
        try:
            document.check_access_rights('read')
            document.check_access_rule('read')
        except AccessError:
            if not access_token or not consteq(document_sudo.access_token, access_token):
                raise
        return document_sudo
```

Reply:
Indeed !

Reply:
Correct !